rfcs/experimental/kernel_templates/README.md

```diff
@@ -0,0 +1,292 @@
+ - Varying numbers of elements to process, from thousands to millions.
+ - Different element types.
+ - Diverse types of input sequences.
+ - Representative data distributions.
+ - Various combinations of other parameters, depending on the signature of the algorithm.
+ - Edge cases (for example, zero-sized sequences).
+
+Due to the unlimited number of possible kernel parameter combinations, it is recommended to:
+ - Always test the most representative configurations
+  (most likely to be used and achieve the best performance).
+ - Randomly select a subset of additional parameter combinations to broaden coverage.
+
+The testing harness should generate or enable tests with limited portability on demand,
+for example through cmake arguments.
+
+## Open Questions
+
+### Name
+
+The name "Kernel Templates" may be misleading because
+these entities more act like algorithms rather than SYCL kernels.
+Probably, a better name is "Algorithm Templates".
+The renaming requires changing the corresponding namespace.
+
+### Specializations and their Differentiation
+
+Currently, the specializations of the algorithms belong to different namespaces.
+Using tags instead of namespaces offers several advantages,
+for example easier dispatching between specializations and avoiding deeply nested namespaces.
+
+### Reporting Global and Local Memory Requirements
+
+Currently, there is no interface that returns the amount of memory to be allocated
+based on the algorithm, its configuration, and input parameters.
+
+Instead, the documentation provides only an estimation, for example see
+https://uxlfoundation.github.io/oneDPL/kernel_templates/single_pass_scan.html#memory-requirements.
+
+The corresponding interfaces would make algorithms safer
+by providing an easy way to get memory requirements and apply fallbacks if needed.
+
+### External Allocation of Global Memory
+
+There are scenarios which benefit from passing pre-allocated temporary memory to an algorithm.
+For example, a user program may manage a reusable memory pool.
+Also, this is especially useful when an algorithm is called in a loop and the allocation overhead
+cannot be amortized by memory pools provided by the device or compiler runtime.
+
+It may be preferable to make this the only option and
+avoid internal global memory allocations altogether.
+This approach eliminates the overhead associated with `host_task`,
+which retains temporary memory until an algorithm completes its asynchronous execution.
+Additionally, it reduces the number of interfaces that need to be supported.
+
+This should be done in conjunction with the
+[Global and Local Memory Requirements](#reporting-global-and-local-memory-requirements) interface.
+
+### Asynchronous Execution
+
+The algorithms return a `sycl::event` but do not accept any input events.
+As a result, they are not fully asynchronous.
+
+It should be investigated whether full asynchrony is desirable,
+and if so, how to implement it correctly.
```

Comment:
What do you mean by "full asynchrony", and why is it only "partial" without input events?

It might be more appropriate to speak about supporting explicit dependency management.

Reply:
Corrected.

I meant that the algorithms cannot be used in asynchronous chains as a non-last point. That limitation does not make them "partially" asynchronous, so thanks for the comment.